lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php

```diff
@@ -157,14 +158,12 @@ public function setInflector(Inflector $inflector): void
     public function loadMetadataForClass(string $className, PersistenceClassMetadata $metadata): void
     {
         if (! $metadata instanceof ClassMetadata) {
-            Deprecation::trigger(
-                'doctrine/orm',
-                'https://github.com/doctrine/orm/pull/249',
-                'Passing an instance of %s to %s is deprecated, please pass a %s instance instead.',
-                $metadata::class,
+            throw new TypeError(sprintf(
+                'Argument #2 passed to %s() must be an instance of %s, %s given.',
```

Comment:
I didn't provide an upgrade note for that one because it's not precisely a realistic scenario to provide anything else than an ORM metadata instance to this method.